lib/ansible/modules/windows/win_domain_user.ps1

```diff
@@ -4,6 +4,109 @@
 
 #Requires -Module Ansible.ModuleUtils.Legacy
 
+Function Test-Credential {
+    param(
+        [String]$Username,
+        [String]$Password,
+        [String]$Domain = $null,
+        $LOGON32_PROVIDER = 0, # Default
+        $LOGON32_LOGON_TYPE = 3 # NetworkLogon
+    )
+    if (($Username.ToCharArray()) -contains [char]'@') {
+        # UserPrincipalName
+        $Username = ($Username -split '@')[0]
+        $Domain = ($Username -split '@')[1]
+    } elseif (($Username.ToCharArray()) -contains [char]'\') {
+        # Pre Win2k Account Name
+        $Username = ($Username -split '\')[0]
+        $Domain = ($Username -split '\')[1]
+    } else {
+        # No domain provided, so maybe local user?
+    }
+    $null = $Domain # Make CI-Check happy...
+
+    # More information about LogonUser at https://docs.microsoft.com/en-us/windows/desktop/api/winbase/nf-winbase-logonusera
+    $platform_util = @'
+using System;
+using System.Runtime.InteropServices;
+using System.ComponentModel;
+using System.Security;
+namespace Ansible
+{
+    public class WinUserPInvoke
+    {
+        const int LOGON32_LOGON_NETWORK           = 3;
```

Comment:
Really isn't a need to provide these contants, if we are always using a network logon and the default provider just pass those values as is an just add a comment.